libevmasm/Assembly.cpp

```diff
@@ -74,6 +74,117 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::addAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(_code.isArray(), "");
+	for (auto const& jsonItem: _code)
+		m_items.emplace_back(loadItemFromJSON(jsonItem));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// During the assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we just ignore exactly these `JUMPDEST`'s.
+		auto const next = std::next(current);
+		if (
+			next != m_items.end() &&
+			current->type() == AssemblyItemType::Tag &&
+			next->type() == AssemblyItemType::Operation &&
+			next->instruction() == Instruction::JUMPDEST
+		)
+			m_items.erase(next);
```

Comment:
Don't you need `current = m_items.erase(next)`? I might also be wrong...

Reply:
It is not needed. The iterator returned by `m_items.erase(next)` would point to the element after next (if it exists). After that, `current` would then be incremented again in the loop and point to the next (next) element. However, I was  not very sure so I just checked the  behaviour with `current = ...`: the import/export test script is not happy with that.